Image class attribute is duplicated

### Astro Info

```block
❯ astro info
Astro                    v5.0.2
Node                     v18.20.3
System                   Linux (x64)
Package Manager          unknown
Output                   static
Adapter                  none
Integrations             none
```



### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

When using the astro:assets `<Image>` if you specify a class in the attributes the class is duplicated because:

https://github.com/withastro/astro/blame/24edead67a6a8a0e78f714ad53fb71792e0b9f1c/packages/astro/components/Image.astro#L60-L61

In that code the attributes.class is added to the end.

So if you `<Image src="/image.png" class="my_class" alt="alt" />`

The result is `<img src="/image.png" alt="alt" class="my_class" class="my_class">`

---

In the blitz reproduction is you run `npm run build`

it builds and then runs `html-validate`

```sh

/home/projects/llqiwjqjgj.github/dist/index.html
  2:241  error  Attribute "class" duplicated  no-dup-attr

✖ 1 problem (1 error, 0 warnings)

More information:
  https://html-validate.org/rules/no-dup-attr.html

```

### What's the expected result?

The expected result is for the class to not be duplicated.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-asg5c8?file=dist%2Findex.html

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/silver-hats-agree.md
+++ b/.changeset/silver-hats-agree.md
@@ -1,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes a bug where the class attribute was rendered twice on the image component

--- a/packages/astro/components/Image.astro
+++ b/packages/astro/components/Image.astro
@@ -47,7 +47,7 @@ if (import.meta.env.DEV) {
 	additionalAttributes['data-image-component'] = 'true';
 }
 
-const attributes = useResponsive
+const { class: className, ...attributes } = useResponsive
 	? applyResponsiveAttributes({
 			layout,
 			image,
@@ -58,4 +58,4 @@ const attributes = useResponsive
 ---
 
 {/* Applying class outside of the spread prevents it from applying unnecessary astro-* classes */}
-<img src={image.src} {...attributes} class={attributes.class} />
+<img src={image.src} {...attributes} class={className} />

--- a/packages/astro/components/Picture.astro
+++ b/packages/astro/components/Picture.astro
@@ -102,7 +102,7 @@ if (fallbackImage.srcSet.values.length > 0) {
 	imgAdditionalAttributes.srcset = fallbackImage.srcSet.attribute;
 }
 
-const attributes = useResponsive
+const { class: className, ...attributes } = useResponsive
 	? applyResponsiveAttributes({
 			layout,
 			image: fallbackImage,
@@ -133,5 +133,5 @@ if (import.meta.env.DEV) {
 		})
 	}
 	{/* Applying class outside of the spread prevents it from applying unnecessary astro-* classes */}
-	<img src={fallbackImage.src} {...attributes} class={attributes.class} />
+	<img src={fallbackImage.src} {...attributes} class={className} />
 </picture>

--- a/packages/astro/test/core-image-layout.test.js
+++ b/packages/astro/test/core-image-layout.test.js
@@ -591,10 +591,9 @@ describe('astro:image:layout', () => {
 			await fixture.build();
 		});
 
-
 		describe('basics', () => {
 			let $;
-			let html
+			let html;
 			before(async () => {
 				html = await fixture.readFile('/index.html');
 				$ = cheerio.load(html);
@@ -680,7 +679,7 @@ describe('astro:image:layout', () => {
 				// We can't use cheerio because it normalises the DOM, so we have to use a regex
 				const matches = html.match(/class="green"/g);
 				assert.equal(matches.length, 1);
-			})
+			});
 
 			it('passes in a parent style', () => {
 				let $img = $('#local-style img');
@@ -697,6 +696,5 @@ describe('astro:image:layout', () => {
 				assert.match(style, /\[data-astro-image\]/);
 			});
 		});
-
 	});
 });

--- a/packages/astro/test/core-image-layout.test.js
+++ b/packages/astro/test/core-image-layout.test.js
@@ -112,6 +112,7 @@ describe('astro:image:layout', () => {
 
 			it('passes in a parent class', () => {
 				let $img = $('#local-class img');
+
 				assert.match($img.attr('class'), /green/);
 			});
 
@@ -575,5 +576,127 @@ describe('astro:image:layout', () => {
 				});
 			});
 		});
+	});
+
+	describe('build', () => {
+		before(async () => {
+			fixture = await loadFixture({
+				root: './fixtures/core-image-layout/',
+				image: {
+					service: testImageService({ foo: 'bar' }),
+					domains: ['avatars.githubusercontent.com'],
+				},
+			});
+
+			await fixture.build();
+		});
+
+
+		describe('basics', () => {
+			let $;
+			let html
+			before(async () => {
+				html = await fixture.readFile('/index.html');
+				$ = cheerio.load(html);
+			});
+
+			it('Adds the <img> tag', () => {
+				let $img = $('#local img');
+				assert.equal($img.length, 1);
+				assert.ok($img.attr('src').startsWith('/_astro'));
+			});
+
+			it('includes lazy loading attributes', () => {
+				let $img = $('#local img');
+				assert.equal($img.attr('loading'), 'lazy');
+				assert.equal($img.attr('decoding'), 'async');
+				assert.equal($img.attr('fetchpriority'), 'auto');
+			});
+
+			it('includes priority loading attributes', () => {
+				let $img = $('#local-priority img');
+				assert.equal($img.attr('loading'), 'eager');
+				assert.equal($img.attr('decoding'), 'sync');
+				assert.equal($img.attr('fetchpriority'), 'high');
+			});
+
+			it('has width and height - no dimensions set', () => {
+				let $img = $('#local img');
+				assert.equal($img.attr('width'), '2316');
+				assert.equal($img.attr('height'), '1544');
+			});
+
+			it('has proper width and height - only width', () => {
+				let $img = $('#local-width img');
+				assert.equal($img.attr('width'), '350');
+				assert.equal($img.attr('height'), '233');
+			});
+
+			it('has proper width and height - only height', () => {
+				let $img = $('#local-height img');
+				assert.equal($img.attr('width'), '300');
+				assert.equal($img.attr('height'), '200');
+			});
+
+			it('has proper width and height - has both width and height', () => {
+				let $img = $('#local-both img');
+				assert.equal($img.attr('width'), '300');
+				assert.equal($img.attr('height'), '400');
+			});
+
+			it('sets the style', () => {
+				let $img = $('#local-both img');
+				assert.match($img.attr('style'), /--w: 300/);
+				assert.match($img.attr('style'), /--h: 400/);
+				assert.equal($img.data('astro-image'), 'responsive');
+			});
+
+			it('sets the style when no dimensions set', () => {
+				let $img = $('#local img');
+				assert.match($img.attr('style'), /--w: 2316/);
+				assert.match($img.attr('style'), /--h: 1544/);
+				assert.equal($img.data('astro-image'), 'responsive');
+			});
+
+			it('sets style for fixed image', () => {
+				let $img = $('#local-fixed img');
+				assert.match($img.attr('style'), /--w: 800/);
+				assert.match($img.attr('style'), /--h: 600/);
+				assert.equal($img.data('astro-image'), 'fixed');
+			});
+
+			it('sets style for full-width image', () => {
+				let $img = $('#local-full-width img');
+				assert.equal($img.data('astro-image'), 'full-width');
+			});
+
+			it('passes in a parent class', () => {
+				let $img = $('#local-class img');
+				assert.equal($img.prop('class'), 'green');
+			});
+
+			it('only passes the class in once', () => {
+				// Check that class="green" only appears once
+				// We can't use cheerio because it normalises the DOM, so we have to use a regex
+				const matches = html.match(/class="green"/g);
+				assert.equal(matches.length, 1);
+			})
+
+			it('passes in a parent style', () => {
+				let $img = $('#local-style img');
+				assert.match($img.attr('style'), /border: 2px red solid/);
+			});
+
+			it('passes in a parent style as an object', () => {
+				let $img = $('#local-style-object img');
+				assert.match($img.attr('style'), /border:2px red solid/);
+			});
+
+			it('injects a style tag', () => {
+				const style = $('style').text();
+				assert.match(style, /\[data-astro-image\]/);
+			});
+		});
+
 	});
 });

